modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/util/ObjectCodeEditorUtil.java

```diff
@@ -366,6 +392,63 @@ private static Map<String, Object> _createCodeEditorElement(
 		).build();
 	}
 
+	private static void _includeRelatedObjectFields(
+			List<Map<String, Object>> codeEditorElements, Locale locale,
+			long objectDefinitionId2,
+			Predicate<ObjectField> objectFieldPredicate)
+		throws PortalException {
+
+		ModelResourcePermission<ObjectDefinition> modelResourcePermission =
+			ModelResourcePermissionRegistryUtil.getModelResourcePermission(
+				ObjectDefinition.class.getName());
+
+		for (ObjectRelationship objectRelationship :
+				ObjectRelationshipLocalServiceUtil.
+					getObjectRelationshipsByObjectDefinitionId2(
+						objectDefinitionId2)) {
+
+			ObjectDefinition objectDefinition1 =
+				ObjectDefinitionLocalServiceUtil.fetchObjectDefinition(
+					objectRelationship.getObjectDefinitionId1());
+
+			if (!modelResourcePermission.contains(
+					PermissionThreadLocal.getPermissionChecker(),
+					objectDefinition1.getObjectDefinitionId(),
+					ActionKeys.VIEW)) {
+
+				continue;
+			}
+
+			ObjectField relationshipObjectField =
+				ObjectFieldLocalServiceUtil.fetchObjectField(
+					objectRelationship.getObjectFieldId2());
+
+			codeEditorElements.add(
+				HashMapBuilder.<String, Object>put(
+					"items",
+					TransformUtil.transform(
+						ListUtil.filter(
+							ObjectFieldLocalServiceUtil.getObjectFields(
+								objectRelationship.getObjectDefinitionId1()),
```

Comment:
Why did you `objectRelationship` instead of `objectDefinition1`?

Reply:
I think it is clear if we use `objectDefinition1`.

Reply:
Done!